src/main/java/uk/gov/hmcts/reform/logging/layout/ReformLoggingLayout.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.hmcts.reform.logging.layout;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.classic.spi.ThrowableProxy;
+import ch.qos.logback.core.CoreConstants;
+import ch.qos.logback.core.LayoutBase;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.hmcts.reform.logging.exception.AbstractLoggingException;
+import uk.gov.hmcts.reform.logging.exception.InvalidExceptionImplementation;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+
+public class ReformLoggingLayout extends LayoutBase<ILoggingEvent> {
+
+    private final Logger log = LoggerFactory.getLogger(ReformLoggingLayout.class);
+
+    private DateTimeFormatter dateFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS");
+
+    /**
+     * By default require thread to be logged.
+     */
+    private boolean requireThread = true;
+
+    /**
+     * By default require an AlertLevel tag to be present for any exception log.
+     */
+    private boolean requireAlertLevel = true;
+
+    public void setDateFormat(String format) {
+        dateFormat = DateTimeFormatter.ofPattern(format);
+    }
+
+    public void setRequireThread(boolean requireThread) {
+        this.requireThread = requireThread;
+    }
+
+    public void setRequireAlertLevel(boolean requireAlertLevel) {
+        this.requireAlertLevel = requireAlertLevel;
+    }
+
+    private void triggerBadImplementationLog(Throwable cause) {
+        Throwable invalid = new InvalidExceptionImplementation("AlertLevel is mandatory as per configuration", cause);
+
+        log.error("Bad implementation of '" + cause.getClass().getCanonicalName() + "' in use", invalid);
+    }
+
+    @Override
+    public String doLayout(ILoggingEvent event) {
+        Instant instant = Instant.ofEpochMilli(event.getTimeStamp());
+        LocalDateTime dateTime = LocalDateTime.ofInstant(instant, ZoneId.systemDefault());
+        StringBuilder log = new StringBuilder(dateFormat.format(dateTime));
+
+        log.append(" ");
+        log.append(String.format("%-5s", event.getLevel().levelStr));
+
+        if (requireThread) {
+            log.append(String.format(" [%s]", event.getThreadName()));
+        }
+
+        int lineNumber = 0;
+
+        if (event.getCallerData().length > 0) {
+            lineNumber = event.getCallerData()[0].getLineNumber();
+        }
+
+        log.append(String.format(" %s:%d: ", event.getLoggerName(), lineNumber));
```

Comment:
should there be `:` after line number?

Reply:
don't know really. usually messages are preceded with colon. now the `logger:line` is more of the question since two colons colliding. can argue about `#`, `@` or just stick with `:`. normally it's not a problem.
furthermore, we can add another flag `includeLineNumber` with default `false`

Reply:
There can also be an alert level between line and actual 'message' as well (the if statement below)

Reply:
With all the confusion on what's being generated, perhaps it would be beneficial to compare entire generated string in tests instead of just substrings?

Reply:
It is tested you even made comments on `ReformLoggingLayoutTest`.

Tell me what's confusing? Layout is representing what would happen behind the scene if we just provide `%d{format} %-5level %thread %logger:%line: %mesage%n` in the logback. Need to provide custom layout as priorities aka alert levels are not something library provides but business invents. And layout as a class itself looks simple.

Reply:
is `2017-10-10 ERROR threadname SomeClass:123: P1 My message.` an expected string then?? (P1 after `:`, before error message)

What's confusing to me is that never in the tests do we see the expected result of what's logged to console. Like the string above.

Reply:
each [test](https://github.com/hmcts/java-logging/pull/15/files#diff-af7186a77430103c0886c8d264d4ca37R66) has a clear string concat to show what message format is expected to be.

where to put `[P1]` is up to discussion. might even ask Jason.